src/streamlink/plugins/vk.py

```diff
@@ -1,78 +1,97 @@
 import logging
 import re
-from html import unescape as html_unescape
 from urllib.parse import parse_qsl, unquote, urlparse
 
 from streamlink.exceptions import NoStreamsError
-from streamlink.plugin import Plugin, pluginmatcher
-from streamlink.plugin.api import useragents
-from streamlink.plugin.api.utils import itertags
+from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.dash import DASHStream
 from streamlink.stream.hls import HLSStream
-from streamlink.stream.http import HTTPStream
-from streamlink.utils.url import update_scheme
 
 log = logging.getLogger(__name__)
 
 
 @pluginmatcher(re.compile(
-    r"https?://(?:\w+\.)?vk\.com/video(?:\?z=video)?(?P<video_id>-?\d*_\d*)"
+    r"https?://(?:\w+\.)?vk\.com/videos?(?:\?z=video)?(?P<video_id>-?\d+_\d+)"
 ))
 @pluginmatcher(re.compile(
-    r"https?://(\w+\.)?vk\.com/videos-?\d*"
+    r"https?://(\w+\.)?vk\.com/.+"
```

Comment:
> https://vk.com/video/for_kids?z=video%2Fpl_-22277933_51053000

some playlist url might result in an IndexError

```
    video_id = self.match.group("video_id")
IndexError: no such group
```

Reply:
That should be resolved with the latest change.